glib/src/gobject/auto/type_module.rs

```diff
@@ -0,0 +1,121 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::{
+    prelude::*, translate::*, EnumValue, FlagsValue, InterfaceInfo, TypeFlags, TypeInfo, TypePlugin,
+};
+use std::fmt;
+
+crate::wrapper! {
+    #[doc(alias = "GTypeModule")]
+    pub struct TypeModule(Object<gobject_ffi::GTypeModule, gobject_ffi::GTypeModuleClass>) @implements TypePlugin;
+
+    match fn {
+        type_ => || gobject_ffi::g_type_module_get_type(),
+    }
+}
+
+impl TypeModule {
+    pub const NONE: Option<&'static TypeModule> = None;
+}
+
+mod sealed {
+    pub trait Sealed {}
+    impl<T: super::IsA<super::TypeModule>> Sealed for T {}
+}
+
+pub trait TypeModuleExt: IsA<TypeModule> + sealed::Sealed + 'static {
+    #[doc(alias = "g_type_module_add_interface")]
+    fn add_interface(
+        &self,
+        instance_type: crate::types::Type,
+        interface_type: crate::types::Type,
+        interface_info: &InterfaceInfo,
+    ) {
+        unsafe {
+            gobject_ffi::g_type_module_add_interface(
+                self.as_ref().to_glib_none().0,
+                instance_type.into_glib(),
+                interface_type.into_glib(),
+                interface_info.to_glib_none().0,
+            );
+        }
+    }
+
+    #[doc(alias = "g_type_module_register_enum")]
+    fn register_enum(&self, name: &str, const_static_values: &EnumValue) -> crate::types::Type {
```

Comment:
The `add()` / `register()` API here, would it make more sense to hide that behind a macro? E.g. the `glib::Enum` macro with some additional configuration?